src/devices/Mcp9808/Mcp9808.cs

```diff
@@ -0,0 +1,206 @@
+    public class Mcp9808 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// MCP9808 I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x18;
+
+        #region prop
+
+        /// <summary>
+        /// MCP9808 Temperature
+        /// </summary>
+        public Temperature Temperature { get => Temperature.FromCelsius(GetTemperature()); }
+
+        private bool _disable;
+
+        /// <summary>
+        /// Disable MCP9808
+        /// </summary>
+        public bool Disabled
+        {
+            get
+            {
+                return _disable;
+            }
+            set
+            {
+                SetShutdown(value);
+                _disable = value;
+            }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Creates a new instance of the MCP9808
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        public Mcp9808(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+
+            Disabled = false;
+
+            if (!Init())
+            {
+                throw new Exception("Unable to identify manufacturer/device id");
+            }
+        }
+
+        /// <summary>
+        /// Checks if the device is a MCP9808
+        /// </summary>
+        /// <returns>True if device has been correctly detected</returns>
+        private bool Init()
+        {
+            if (Read16(Register8.MCP_MANUF_ID) != 0x0054)
+            {
+                return false;
+            }
+
+            if (Read16(Register8.MCP_DEVICE_ID) != 0x0400)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Return the internal resolution register
+        /// </summary>
+        /// <returns>Resolution setting</returns>
+        public byte GetResolution()
+        {
+            return Read8(Register8.MCP_RESOLUTION);
+        }
+
+        /// <summary>
+        /// Wakes-up the device
+        /// </summary>
+        public void Wake()
+        {
+            SetShutdown(false);
+
+            // Sleep 250 ms, which is the typical temperature conversion time for the highest resolution (page 3 in datasheet, tCONV)
+            Thread.Sleep(250);
+        }
+
+        /// <summary>
+        /// Shuts down the device
+        /// </summary>
+        public void Shutdown()
+        {
+            SetShutdown(true);
+        }
+
+        /// <summary>
+        /// Read MCP9808 Temperature (℃)
+        /// </summary>
+        /// <returns>Temperature</returns>
+        private double GetTemperature()
+        {
+            ushort value = Read16(Register8.MCP_AMBIENT_TEMP);
+
+            if (value == 0xFFFF)
+            {
+                return double.NaN;
+            }
+
+            double temp = value & 0x0FFF;
+            temp /= 16.0;
+            if ((value & 0x1000) != 0)
```

Comment:
Do you mind also adding a comment into what does this check mean? Is that to catch an overflow?

Reply:
I just copied it from an Arduino library for the MCP9808. I assume there's some reason to it in the datasheet, but didn't look into that.

Reply:
I see, now that you mention that, can you check if that Arduino library has permisive license? If we are porting code from an existing open source library, we are required to make sure that the ported code is compatible with our license. If you point me to the source of that library I can take a look if it does.

Reply:
I didn't actually port their library, I just took these 3 lines and re-wrote them to C# (plus the part with NaN). But here it is: https://github.com/adafruit/Adafruit_MCP9808_Library

Reply:
I see, that's ok though since it looks like they are under BSD license which means that is ok 😄